src/fdcache.cpp

```diff
@@ -482,13 +501,270 @@ int PageList::GetUnloadedPages(fdpage_list_t& unloaded_list, off_t start, off_t
       // merge to before page
       riter->bytes += page_size;
     }else{
-      fdpage page(page_start, page_size, false);
+      fdpage page(page_start, page_size, false, false);
       unloaded_list.push_back(page);
     }
   }
   return unloaded_list.size();
 }
 
+// [NOTE]
+// This method is called in advance when mixing POST and COPY in multi-part upload.
+// The minimum size of each part must be 5 MB, and the data area below this must be
+// downloaded from S3.
+// This method checks the current PageList status and returns the area that needs
+// to be downloaded so that each part is at least 5 MB.
+//
+bool PageList::GetLoadPageListForMultipartUpload(fdpage_list_t& dlpages)
```

Comment:
I will try to improve the test.
It takes some consideration, because it needs to be written to the area that is not loaded, but it seems to be feasible.

Reply:
Agree this requires some work but I think the unit test will be better than trying to write integration tests for this behavior.